src/jobs/jobs.controller.ts

```diff
@@ -13,21 +13,27 @@ export class JobsController {
 
   @Get()
   async findAll() {
-    const jobs = await this.jobsService.findAll()
-    if (!jobs) {
-      throw new InternalServerErrorException(
-        `Error occurred while fetching jobs`,
+    try {
+      return await this.jobsService.findAll()
+    } catch (e) {
+      throw new BadGatewayException(
+        e.message || 'Error occurred while fetching jobs',
       )
     }
-    return jobs
   }
 
   @Get(':id')
   async findOne(@Param('id') id: string) {
-    const job = await this.jobsService.findOne(id)
-    if (!job) {
-      throw new NotFoundException(`Job with id ${id} not found`)
+    try {
+      const job = await this.jobsService.findOne(id)
```

Comment:
Didn't you say that Axios throws on a 404 response? 

Won't this just throw, and then get caught and then respond with the 502 bag gateway below? 🤔

Reply:
Looks like it does 😕

https://github.com/axios/axios/blob/d032edda08948f3f613ff34f32af49cad3aa74db/lib/defaults/index.js#L148